src/parse.c

```diff
@@ -63,6 +63,45 @@ int missing_ids_found;
  * Loading and error handling
  ****************************************************/
 
+static char *
+get_error_context(yaml_parser_t parser, GError **error)
+{
+    GString *message = NULL;
+    unsigned char* line = parser.buffer.pointer;
+    unsigned char* current = line;
+    int i;
+
+    message = g_string_sized_new(200);
+
+    while (current > parser.buffer.start) {
+        current--;
+        if (*current == '\n') {
+            line = current + 2;
+            break;
+        }
+    }
+    if (current <= parser.buffer.start)
+        line = parser.buffer.start;
+    current = line + 1;
+    while (current <= parser.buffer.end) {
+        if (*current == '\n') {
+            *current = '\0';
+            break;
+        }
+        current++;
+    }
+
+    g_string_append_printf(message, "%s\n", line);
+
+    for (i = 0;
+         (parser.problem_mark.column > 0 && i < (parser.problem_mark.column - 1));
+         i++)
+        g_string_append_printf(message, " ");
+    g_string_append_printf(message, "^");
+
```

Comment:
Irritatingly, this isn't printing the invalid tab character for me.

Input:
```
networks:
        hi: {}
```
(where the second line is indented with a tab)

Output:
```
Invalid YAML at testroot//etc/netplan/a.yaml line 1 column 0: tabs are not allowed for indent:
hi: {}
^
```
The caret is pointing at the wrong thing - it's somehow stripped/missed the actual tab character. This makes it look like 'hi' is the problem, rather than the tab.